File: app/src/androidTest/java/com/danielmaia/desafioandroid/MainActivityTest.java
package com.danielmaia.desafioandroid;

import androidx.test.espresso.matcher.ViewMatchers;
import androidx.test.rule.ActivityTestRule;
import androidx.test.runner.AndroidJUnit4;

import com.danielmaia.desafioandroid.view.MainActivity;
import com.danielmaia.desafioandroid.R;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {

    @Rule
    public ActivityTestRule<MainActivity> mActivityRule =
            new ActivityTestRule<>(MainActivity.class, false, true);

    @Test
    public void checkInitialStateMainActivity() {
        onView(ViewMatchers.withId(R.id.toolbar)).check(matches(isDisplayed()));
    }

}

File: app/src/androidTest/java/com/pedrenrique/githubapp/features/MainActivityTest.kt
package com.pedrenrique.githubapp.features

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.rule.ActivityTestRule
import com.pedrenrique.githubapp.R
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith


@RunWith(AndroidJUnit4::class)
class MainActivityTest {

    private val activityTestRule = ActivityTestRule(MainActivity::class.java, false, false)

    @Before
    fun setUp() {
        activityTestRule.launchActivity(null)
    }

    @Test
    fun testDefaultNavigation() {
        onView(withId(R.id.rvRepositories))
            .check(matches(isDisplayed()))
    }
}